common/src/main/java/whocraft/tardis_refined/registry/CustomRegistry.java

```diff
@@ -0,0 +1,41 @@
+package whocraft.tardis_refined.registry;
+
+import com.mojang.serialization.Codec;
+import dev.architectury.injectables.annotations.ExpectPlatform;
+import net.minecraft.core.Registry;
+import net.minecraft.resources.ResourceKey;
+import net.minecraft.resources.ResourceLocation;
+
+import java.util.Collection;
+import java.util.Optional;
+import java.util.Set;
+import java.util.function.Supplier;
+
+public abstract class CustomRegistry<T> {
```

Comment:
This implementation looks to be quite different from upstream 20.2. Any reason for this refactor?